Can't build docker image.

1. git clone https://github.com/flant/shell-operator.git
2. docker build .
```
Sending build context to Docker daemon  519.2kB
Step 1/14 : FROM golang:1.12-alpine3.9
 ---> bcc226f44d8e
Step 2/14 : ARG appVersion=latest
 ---> Using cache
 ---> 1c999505e352
Step 3/14 : RUN apk --no-cache add git ca-certificates
 ---> Using cache
 ---> 12fa79d1bd54
Step 4/14 : ADD . /go/src/github.com/flant/shell-operator
 ---> fad6ecb2db1d
Step 5/14 : RUN go get -d github.com/flant/shell-operator/...
 ---> Running in f15e519054c5
Removing intermediate container f15e519054c5
 ---> 7e45c102ab06
Step 6/14 : WORKDIR /go/src/github.com/flant/shell-operator
 ---> Running in 57f0f8f79e0a
Removing intermediate container 57f0f8f79e0a
 ---> 8a0a147b24aa
Step 7/14 : RUN CGO_ENABLED=0 GOOS=linux go build -ldflags="-s -w -X 'github.com/flant/shell-operator/pkg/app.Version=$appVersion'" -o shell-operator ./cmd/shell-operator
 ---> Running in 2e4b4481f957
# github.com/googleapis/gnostic/extensions
../../googleapis/gnostic/extensions/extensions.go:39:14: undefined: ExtensionHandlerRequest
../../googleapis/gnostic/extensions/extensions.go:50:15: undefined: ExtensionHandlerResponse
# k8s.io/client-go/transport
/go/src/k8s.io/client-go/transport/round_trippers.go:70:11: cannot convert klog.V(9) (type klog.Verbose) to type bool
/go/src/k8s.io/client-go/transport/round_trippers.go:72:11: cannot convert klog.V(8) (type klog.Verbose) to type bool
/go/src/k8s.io/client-go/transport/round_trippers.go:74:11: cannot convert klog.V(7) (type klog.Verbose) to type bool
/go/src/k8s.io/client-go/transport/round_trippers.go:76:11: cannot convert klog.V(6) (type klog.Verbose) to type bool
# k8s.io/utils/trace
/go/src/k8s.io/utils/trace/trace.go:100:57: invalid operation: stepThreshold == 0 || stepDuration > stepThreshold || klog.V(4) (mismatched types bool and klog.Verbose)
/go/src/k8s.io/utils/trace/trace.go:112:56: invalid operation: stepThreshold == 0 || stepDuration > stepThreshold || klog.V(4) (mismatched types bool and klog.Verbose)
The command '/bin/sh -c CGO_ENABLED=0 GOOS=linux go build -ldflags="-s -w -X 'github.com/flant/shell-operator/pkg/app.Version=$appVersion'" -o shell-operator ./cmd/shell-operator' returned a non-zero code: 2
```
I think this commit https://github.com/googleapis/gnostic/commit/51de230aea969b1a9d9de04e8f762f2d82c4b98a had huge impact to this process.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -4,19 +4,18 @@ version: 2.1
 executors:
   golang:
     docker:
-    - image: circleci/golang:1.11
+    - image: circleci/golang:1.12
 
 jobs:
   test:
     executor: golang
-    working_directory: /go/src/github.com/flant/shell-operator
+    working_directory: /tmp/flant/shell-operator
     steps:
       - checkout
-      - run: go get -v -t -d ./...
+      - run: go mod download
       - run: go build -v ./cmd/shell-operator
+      - run: go test -race -v ./...
       # TODO add vet, lint, speller
-      # TODO fix tests
-      #- run: go test -v ./...
 
   build_image_branch:
     machine: true

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,0 +1,2 @@
+Dockerfile
+Dockerfile-alpine3.9

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,9 +1,13 @@
 FROM golang:1.12-alpine3.9
 ARG appVersion=latest
 RUN apk --no-cache add git ca-certificates
-ADD . /go/src/github.com/flant/shell-operator
-RUN go get -d github.com/flant/shell-operator/...
-WORKDIR /go/src/github.com/flant/shell-operator
+
+# Cache-friendly download of go dependencies.
+ADD go.mod go.sum /src/shell-operator/
+WORKDIR /src/shell-operator
+RUN go mod download
+
+ADD . /src/shell-operator
 RUN CGO_ENABLED=0 GOOS=linux go build -ldflags="-s -w -X 'github.com/flant/shell-operator/pkg/app.Version=$appVersion'" -o shell-operator ./cmd/shell-operator
 
 FROM ubuntu:18.04
@@ -13,7 +17,7 @@ RUN apt-get update && \
     wget https://storage.googleapis.com/kubernetes-release/release/v1.13.5/bin/linux/amd64/kubectl -O /bin/kubectl && \
     chmod +x /bin/kubectl && \
     mkdir /hooks
-COPY --from=0 /go/src/github.com/flant/shell-operator/shell-operator /
+COPY --from=0 /src/shell-operator /
 WORKDIR /
 ENV SHELL_OPERATOR_WORKING_DIR /hooks
 ENTRYPOINT ["/shell-operator"]

--- a/Dockerfile-alpine3.9
+++ b/Dockerfile-alpine3.9
@@ -1,9 +1,13 @@
 FROM golang:1.12-alpine3.9
 ARG appVersion=latest
 RUN apk --no-cache add git ca-certificates
-ADD . /go/src/github.com/flant/shell-operator
-RUN go get -d github.com/flant/shell-operator/...
-WORKDIR /go/src/github.com/flant/shell-operator
+
+# Cache-friendly download of go dependencies.
+ADD go.mod go.sum /src/shell-operator/
+WORKDIR /src/shell-operator
+RUN go mod download
+
+ADD . /src/shell-operator
 RUN CGO_ENABLED=0 GOOS=linux go build -ldflags="-s -w -X 'github.com/flant/shell-operator/pkg/app.Version=$appVersion'" -o shell-operator ./cmd/shell-operator
 
 FROM alpine:3.9
@@ -11,7 +15,7 @@ RUN apk --no-cache add ca-certificates jq bash && \
     wget https://storage.googleapis.com/kubernetes-release/release/v1.13.5/bin/linux/amd64/kubectl -O /bin/kubectl && \
     chmod +x /bin/kubectl && \
     mkdir /hooks
-COPY --from=0 /go/src/github.com/flant/shell-operator/shell-operator /
+COPY --from=0 /src/shell-operator /
 WORKDIR /
 ENV SHELL_OPERATOR_WORKING_DIR /hooks
 ENTRYPOINT ["/shell-operator"]

--- a/pkg/queue/queue.go
+++ b/pkg/queue/queue.go
@@ -41,7 +41,9 @@ func (q *Queue) Add(task interface{}) {
 
 // Peek examines the first element (get without delete).
 func (q *Queue) Peek() (task interface{}, err error) {
-	if q.IsEmpty() {
+	q.m.Lock()
+	defer q.m.Unlock()
+	if q.isEmpty() {
 		return nil, err
 	}
 	return q.items[0], err

